### ISSUE TITLE
page 2 of search results are broken (passing count: 0)

### ISSUE THREAD

**Issue by tofumatt** [mozilla/addons-frontend#1401]
The second page of search results are broken: https://addons-dev.allizom.org/en-GB/firefox/search/?page=2&q=test

Not sure why it's happening or when it started, but threw me for a loop on a branch when I thought I messed something up, only to see master was broken...

**Comment by muffinresearch** [comment]
Looks like this problem was introduced by https://github.com/mozilla/addons-frontend/commit/9db4c876d15b3d8c76bc08eec2a7755af1b1fb1c

**Comment by muffinresearch** [comment]
@tofumatt I'm downgrading this to p2 since it's not affecting anything in production.

**Comment by mstriemer** [comment]
I haven't tested this out but it looks like this could be the cause:

```js
export function filtersMatch(filters, otherFilters) {
  return Object.keys(filters).every((key) => (
    filters[key] === otherFilters[key]
  ));
}
```

When you're on the first page `filters` doesn't include a `page` key. But on the next page it will.

This should either check that both objects have the same keys before the current `every()` call or spread the keys together before iterating with `[...Object.keys(filters), ...Object.keys('otherFilters')].every(...)`.

**Comment by mstriemer** [comment]
Actually it looks like that function just checks `filters` not page. Either way there's a potential bug in there.

**Comment by tofumatt** [comment]
Indeed, thanks for that!

I’ve got it working now and am gonna improve both the checks and the tests
around it so this doesn’t happen again. Should have a PR in a bit.

**Comment by vcarciu** [comment]
If you press the Next button for going the second page in search results it will works , but if you directly open the link to second page , will return Not Found : https://addons-dev.allizom.org/en-US/android/search/?page=2&q=Fire

**Comment by tofumatt** [comment]
Actually, it looks like ANY call to search that makes an API request fails on the server. Checking into this now. (eg https://addons-dev.allizom.org/en-US/android/search/?q=Fire hangs and fails)

**Comment by tofumatt** [comment]
Commit 593b5742a76449879a344a6114ef3ad272253672 seems to be the culprit; commits before it load pages that make a `callApi` request from the server without issue. Guessing there is a difference in the node versus client API on `fetch` or something, checking now.

**Comment by tofumatt** [comment]
Turns out this is a regression from 593b5742a76449879a344a6114ef3ad272253672 and because it's somewhat unrelated I'm gonna open it as it's own issue to not confuse things, as it's larger in scope than this issue makes it out to be.

While these URLs don't work the original bug has been fixed and the issue is loading these pages directly (eg making a server request to node rather than running the code and request from the client).

I've opened #1726 to track this issue as it's larger in scope.
